x-pack/legacy/plugins/uptime/public/state/actions/types.ts

```diff
@@ -5,11 +5,21 @@
  */
 
 import { Action } from 'redux-actions';
+import { IHttpFetchError } from '../../../../../../../target/types/core/public/http';
 
-export interface AsyncAction {
-  get: (payload?: any) => Action<any>;
-  success: (payload?: any) => Action<any>;
-  fail: (payload?: any) => Action<any>;
+export interface AsyncAction<Payload, SuccessPayload> {
+  get: (payload: Payload) => Action<Payload>;
+  success: (payload: SuccessPayload) => Action<SuccessPayload>;
+  fail: (payload: IHttpFetchError) => Action<IHttpFetchError>;
+}
+export interface AsyncAction1<Payload, SuccessPayload> {
```

Comment:
Is this needed?

Reply:
yes it's need to have two types of functions.